superdesk/publish/publish_queue.py

```diff
@@ -116,6 +130,32 @@ def _set_queue_state(self, updates, original):
             else updates.get("state") or QueueState.PENDING.value
         )
 
+    def is_corrected_document(self, doc):
+        if not (doc.get("item_id") and doc.get("item_version", "version")):
+            return
+        article = get_resource_service("published").find_one(
+            req=None, guid=doc["item_id"], _current_version=doc.get("item_version", "version")
+        )
+        if not article:
+            return
+        return doc.get("publishing_action") == "corrected" or article.get("operation") == "being_corrected"
+
+    def resend_association_items(self, doc):
```

Comment:
I'm wondering why it's actually done in publish queue, seems like we're splitting the logic in 2 places, but why?
seems like all the info is already there, but it goes in publish queue to call the enqueue service eventually

Reply:
We have resend option in publish_queue also and publish_queue do not use the enqueue service, in all other scenario enqueue service is called except resend option in publish_queue.

Reply:
resend action we have now in the ui has nothing to do with this really, doesn't it?

Reply:
when we do resend item(3 dot menu) or publish then first enqueue service is called after that publish queue but when we resend (from publish queue list) then directly publish queue called.
that is the reason we implemented that in publish queue instead of enqueue service as in all the cases whether it is resend items (3 dot menu) or publish or send correction or resend (publish queue list) publish queue will always be called.
after that, if we have config and associated items then using enqueue service publish method we will resend association as well.

Reply:
I think this shouldn't change the behaviour of resend called from publish queue, that one should always just resend that very item you click on, and not its associations

Reply:
So that means resend (from publish queue list) will resend only that main item but not the associated items?

Reply:
right